tools/build_config/Jenkinsfile

```diff
@@ -403,6 +458,53 @@ pipeline {
         )
       }
     }
+
+
+    stage('Update docs') {
+      when {
+        expression {env.RELEASE_TYPE == "nightly"}
+      }
+      steps {
+        withCredentials([string(credentialsId: 'GitHub_API_Token',
+                                variable: 'api_token')]) {
+          script {
+            if (isUnix()) {
```

Comment:
Similarly I'm not sure these large code blocks should be contained in the main Jenkinsfile, as there is no way for a developer to run the same operation outside of Jenkins. 

I understand this is a more subtle case to the build above, the git config and remote set-url are clearly specific to the Jenkinsfile setting up the git environment.